### PR TITLE
[Security] Add Support for DotNet Attributes Part 2

### PR DESCRIPTION
**Pull request by tj-devel709**

### CODE REVIEW

File: src/Security/SecIdentity2.cs [hunk 1]
```diff
@@ -14,13 +14,20 @@
...
+using System.Runtime.Versioning;
 using ObjCRuntime;
 using Foundation;
 using CoreFoundation;
 
 namespace Security {
 
-	[TV (12,0), Mac (10,14), iOS (12,0), Watch (5,0)]
+#if !NET
+	[iOS (12,0), TV (12,0), Mac (10,14), Watch (5,0)]
```
Comment (mandel-macaque) [inline src/Security/SecIdentity2.cs:25]: AS with the other PRs and as requested by @spouliot  lets not change the original.

File: src/Security/SecIdentity2.cs [hunk 2]
```diff
@@ -70,7 +77,13 @@ public SecIdentity2 (SecIdentity identity, params SecCertificate [] certificates
 			}
 		}
 
-		[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+#if !NET
+		[iOS (13,0), TV (13,0), Mac (10,15), Watch (6,0)]
```
Comment (mandel-macaque) [inline src/Security/SecIdentity2.cs:81]: same.

File: src/Security/SecIdentity2.cs [hunk 3]
```diff
@@ -86,7 +99,13 @@ static void TrampolineAccessCertificates (IntPtr block, IntPtr cert)
 				del (new SecCertificate2 (cert, false));
 		}
 
-		[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+#if !NET
+		[iOS (13,0), TV (13,0), Mac (10,15), Watch (6,0)]
```
Comment (mandel-macaque) [inline src/Security/SecIdentity2.cs:103]: same.

File: src/Security/SecPolicy.cs [hunk 4]
```diff
@@ -17,33 +18,63 @@ namespace Security {
...
+#else
+		[SupportedOSPlatform ("macos10.9")]
+#endif
 		[DllImport (Constants.SecurityLibrary)]
 		extern static IntPtr /* __nullable SecPolicyRef */ SecPolicyCreateRevocation (/* CFOptionFlags */ nuint revocationFlags);
 
-		[Mac (10,9)][iOS (7,0)]
+#if !NET
+		[iOS (7,0)]
+		[Mac (10,9)]
```
Comment (mandel-macaque) [inline src/Security/SecPolicy.cs:50]: same.

File: src/Security/SecPolicy.cs [hunk 5]
```diff
@@ -17,33 +18,63 @@ namespace Security {
...
 		static public SecPolicy CreateRevocationPolicy (SecRevocation revocationFlags)
 		{
 			var policy = SecPolicyCreateRevocation ((nuint)(ulong) revocationFlags);
 			return policy == IntPtr.Zero ? null : new SecPolicy (policy, true);
 		}
 
-		[Mac (10,9)][iOS (7,0)]
+#if !NET
+		[iOS (7,0)]
+		[Mac (10,9)]
```
Comment (mandel-macaque) [inline src/Security/SecPolicy.cs:63]: same.

File: src/Security/SecPolicy.cs [hunk 6]
```diff
@@ -17,33 +18,63 @@ namespace Security {
...
+		[SupportedOSPlatform ("ios7.0")]
+		[SupportedOSPlatform ("macos10.9")]
+#endif
 		[DllImport (Constants.SecurityLibrary)]
 		extern static IntPtr /* __nullable SecPolicyRef */ SecPolicyCreateWithProperties (IntPtr /* CFTypeRef */ policyIdentifier, IntPtr /* CFDictionaryRef */ properties);
 
-		[Mac (10,9)][iOS (7,0)]
+#if !NET
+		[iOS (7,0)]
+		[Mac (10,9)]
```
Comment (mandel-macaque) [inline src/Security/SecPolicy.cs:73]: same.

File: src/Security/SecProtocolMetadata.cs [hunk 7]
```diff
@@ -17,7 +18,13 @@
 using dispatch_queue_t=System.IntPtr;
 
 namespace Security {
-	[TV (12,0), Mac (10,14), iOS (12,0), Watch (5,0)]
+#if !NET
+		[iOS (12,0), TV (12,0), Mac (10,14), Watch (5,0)]
```
Comment (mandel-macaque) [inline src/Security/SecProtocolMetadata.cs:22]: same.

File: src/Security/SecProtocolMetadata.cs [hunk 8]
```diff
@@ -36,44 +43,120 @@ public class SecProtocolMetadata : NativeObject {
...
 
-		[Deprecated (PlatformName.MacOSX, 10,15, message: "Use 'NegotiatedTlsProtocolVersion' instead.")]
-		[Deprecated (PlatformName.iOS, 13,0, message: "Use 'NegotiatedTlsProtocolVersion' instead.")]
-		[Deprecated (PlatformName.WatchOS, 6,0, message: "Use 'NegotiatedTlsProtocolVersion' instead.")]
-		[Deprecated (PlatformName.TvOS, 13,0, message: "Use 'NegotiatedTlsProtocolVersion' instead.")]
+#if !NET
+		[Deprecated (PlatformName.iOS, 13, 0, message: "Use 'NegotiatedTlsProtocolVersion' instead.")]
+		[Deprecated (PlatformName.TvOS, 13, 0, message: "Use 'NegotiatedTlsProtocolVersion' instead.")] 
+		[Deprecated (PlatformName.MacOSX, 10, 15, message: "Use 'NegotiatedTlsProtocolVersion' instead.")]
+		[Deprecated (PlatformName.WatchOS, 6, 0, message:  "Use 'NegotiatedTlsProtocolVersion' instead.")]
```
Comment (mandel-macaque) [inline src/Security/SecProtocolMetadata.cs:50]: same.

File: src/Security/SecProtocolMetadata.cs [hunk 9]
```diff
@@ -36,44 +43,120 @@ public class SecProtocolMetadata : NativeObject {
...
+#else
+		[UnsupportedOSPlatform ("ios13.0")]
+		[UnsupportedOSPlatform ("tvos13.0")]
+		[UnsupportedOSPlatform ("macos10.15")]
+#if IOS
+		[Obsolete ("Starting with ios13.0 ", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif TVOS
+		[Obsolete ("Starting with tvos13.0 ", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif MONOMAC
+		[Obsolete ("Starting with macos10.15 ", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```
Comment (mandel-macaque) [inline src/Security/SecProtocolMetadata.cs:137]: Not a valid message. Correct message would be 'Use "sec_protocol_metadata_get_negotiated_tls_ciphersuite" instead.'

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
Please undo the changes done in the progiial attributes before the PR.

**Review by rolfbjarne — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

File: src/Security/SecProtocolOptions.cs [hunk 10]
```diff
@@ -36,116 +43,319 @@ public void SetLocalIdentity (SecIdentity2 identity)
...
 		[Deprecated (PlatformName.WatchOS, 6,0, message: "Use 'SetTlsMinVersion (TlsProtocolVersion)' instead.")]
 		[Deprecated (PlatformName.TvOS, 13,0, message: "Use 'SetTlsMinVersion (TlsProtocolVersion)' instead.")]
+#else
+		[UnsupportedOSPlatform ("ios13.0")]
+		[UnsupportedOSPlatform ("tvos13.0")]
+		[UnsupportedOSPlatform ("macos10.15")]
+#if IOS
+		[Obsolete ("Starting with ios13.0 use 'SetTlsMinVersion (TlsProtocolVersion)' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif TVOS
+		[Obsolete ("Starting with tvos13.0 use 'SetTlsMinVersion (TlsProtocolVersion)' instead.' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```
Comment (mandel-macaque) [inline src/Security/SecProtocolOptions.cs:176]:
```suggestion
		[Obsolete ("Starting with tvos13.0 use 'SetTlsMinVersion (TlsProtocolVersion)' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```

File: src/Security/SecProtocolOptions.cs [hunk 11]
```diff
@@ -36,116 +43,319 @@ public void SetLocalIdentity (SecIdentity2 identity)
...
 		[Unavailable (PlatformName.MacCatalyst)]
+#else
+		[UnsupportedOSPlatform ("ios13.0")]
+		[UnsupportedOSPlatform ("tvos13.0")]
+		[UnsupportedOSPlatform ("macos10.15")]
+		[UnsupportedOSPlatform ("maccatalyst")]
+#if IOS
+		[Obsolete ("Starting with ios13.0 use 'SetTlsMinVersion (TlsProtocolVersion)' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif TVOS
+		[Obsolete ("Starting with tvos13.0 use 'SetTlsMinVersion (TlsProtocolVersion)' instead.' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```
Comment (mandel-macaque) [inline src/Security/SecProtocolOptions.cs:198]:
```suggestion
		[Obsolete ("Starting with tvos13.0 use 'SetTlsMinVersion (TlsProtocolVersion)' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```

File: src/Security/SecProtocolOptions.cs [hunk 12]
```diff
@@ -168,17 +378,43 @@ public void SetTlsServerName (string serverName)
...
 		[Deprecated (PlatformName.WatchOS, 6,0, message: "Use non-DHE cipher suites instead.")]
 		[Deprecated (PlatformName.TvOS, 13,0, message: "Use non-DHE cipher suites instead.")]
+#else
+		[UnsupportedOSPlatform ("ios13.0")]
+		[UnsupportedOSPlatform ("tvos13.0")]
+		[UnsupportedOSPlatform ("macos10.15")]
+#if IOS
+		[Obsolete ("Starting with ios13.0 use non-DHE cipher suites instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif TVOS
+		[Obsolete ("Starting with tvos13.0 use non-DHE cipher suites instead.' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```
Comment (mandel-macaque) [inline src/Security/SecProtocolOptions.cs:393]:
```suggestion
		[Obsolete ("Starting with tvos13.0 use non-DHE cipher suites instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```

File: src/Security/SecProtocolOptions.cs [hunk 13]
```diff
@@ -168,17 +378,43 @@ public void SetTlsServerName (string serverName)
...
```
Comment (mandel-macaque) [inline src/Security/SecProtocolOptions.cs:413]:
```suggestion
		[Obsolete ("Starting with tvos13.0 use non-DHE cipher suites instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```

**Review by mandel-macaque — APPROVED** [review]
Approving after I fixed the strings.

**Comment by tj-devel709** [comment]
Thanks, @mandel-macaque!!

**Comment by tj-devel709** [comment]
Unrelated Test failures: https://github.com/xamarin/maccore/issues/2443 and https://github.com/xamarin/maccore/issues/2414

**Pull request merged**